test/index.js

```diff
@@ -34,7 +37,29 @@ const eq = actual => expected => {
   strictEqual (Z.equals (actual, expected), true);
 };
 
-const testModule = (expecteds, path, options) => {
+const Test = description => input => output => comparison => ({
+  description,
+  test: {
+    lines: {
+      input,
+      output,
+    },
+    comparison,
+  },
+});
+
+const dependencies = {
+  Test,
+  Line,
+  Incorrect,
+  Correct,
+  Failure,
+  Success,
+};
+
+const testModule = (module, path, options) => {
+  const expecteds = module (dependencies);
```

Comment:
The dependency injection looks a little bit like reinventing modules. Couldn't you just use `import {Test, Line} from './deps.js'` in every test file?

Reply:
It allows for easily running the tests differently by supplying a different `Test` function. I'll mark this as resolved.